.github/workflows/code-analysis.yml

```diff
@@ -0,0 +1,31 @@
+name: PHP Composer
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  php-stan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install packages and run PHP Stan
+      uses: actions/checkout@v2
+    - run: composer install
+    - run: sudo cp vendor/prestashop/php-dev-tools/phpstan/ps-module-extension.neon vendor/prestashop/php-dev-tools/phpstan/extension.neon
+    - run: sudo cp vendor/prestashop/php-dev-tools/templates/phpstan/phpstan.neon phpstan.neon
+    - run: sudo chown -R $USER:$USER phpstan.neon
+    - run: sudo echo '  excludes_analyse:' >> phpstan.neon
+    - run: sudo echo '    - %currentWorkingDirectory%/vendor/*' >> phpstan.neon
```

Comment:
Configurations should be made part of the repo so that developers can also run this locally